core-rust/state-manager/src/store/rocks_db.rs

```diff
@@ -284,6 +317,12 @@ impl RocksDBStore {
             .unwrap()
             .map(|pinnable_slice| scrypto_decode(pinnable_slice.as_ref()).unwrap())
     }
+
+    fn put<T: ScryptoEncode>(&self, cf: &RocksDBColumnFamily, key: &[u8], value: &T) {
```

Comment:
Why are we exposing a generic `put` here? It feels quite dangerous.

I'd rather all updates go through a batch for atomicity.